File: socialuni-server/socialuni-im-api/src/main/java/com/socialuni/social/im/api/feign/SocialuniImUserAPI.java
package com.socialuni.social.im.api.feign;


import com.socialuni.social.common.api.model.ResultRO;
import com.socialuni.social.common.api.model.user.SocialuniUserIdQO;
import com.socialuni.social.im.api.model.QO.SocialuniChatQueryQO;
import com.socialuni.social.im.api.model.QO.chat.ChatReadVO;
import com.socialuni.social.im.api.model.QO.chat.ChatRemoveVO;
import com.socialuni.social.im.api.model.QO.chat.OpenChatVO;
import com.socialuni.social.im.api.model.RO.ChatRO;
import com.socialuni.social.im.api.model.RO.SocialuniImMineUserDetailRO;
import com.socialuni.social.im.api.model.RO.SocialuniImUserDetailRO;
import io.swagger.v3.oas.annotations.tags.Tag;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;

import javax.validation.Valid;
import javax.validation.constraints.NotNull;
import java.util.List;

/**
 * @author qinkaiyuan
 * @date 2018-11-18 20:45
 */

//@RequestMapping("socialuni/chat")
@FeignClient(name = "imUser", url = "${socialuni.central-server-url:https://api.socialuni.cn}", path = "socialuni/imUser")
@Tag(name = "消息模块/会话模块",description = "暂未支持")
public interface SocialuniImUserAPI {
    @PostMapping("getImUserDetail")
    ResultRO<SocialuniImUserDetailRO> getImUserDetail(@RequestBody @Valid SocialuniUserIdQO socialuniUserIdQO);

    @PostMapping("getImMineUser")
    ResultRO<SocialuniImMineUserDetailRO> getImMineUser();

    @PostMapping("closeStrangerMsg")
    ResultRO<SocialuniImMineUserDetailRO> closeStrangerMsg();

    @PostMapping("openStrangerMsg")
    ResultRO<SocialuniImMineUserDetailRO> openStrangerMsg();
}

File: socialuni-js/packages/socialuni-im/socialuni-im-api/src/api/SocialuniImUserAPI.ts
import FriendAddQO from "../model/QO/firend/FriendAddQO";
import socialuniImRequest from "./socialuniImRequest";
import SocialuniFriendApplyUserRO from "../model/RO/SocialuniFriendApplyUserRO";
import SocialuniUserIdQO from 'socialuni-api-base/src/model/QO/user/SocialuniUserIdQO'
import SocialuniImMineUserDetailRO from '../model/RO/SocialuniImMineUserDetailRO'
import SocialuniImUserDetailRO from '../model/RO/SocialuniImUserDetailRO'

export default class SocialuniImUserAPI {
    static getImUserDetailAPI(userIdQO: SocialuniUserIdQO) {
        return socialuniImRequest.post<SocialuniImUserDetailRO>('socialuni/imUser/getImUserDetail', userIdQO)
    }

    static getImMineUser() {
        return socialuniImRequest.post<SocialuniImMineUserDetailRO>('socialuni/imUser/getImMineUser')
    }

    static closeStrangerMsg() {
        return socialuniImRequest.post<SocialuniImMineUserDetailRO>('socialuni/imUser/closeStrangerMsg')
    }

    static openStrangerMsg() {
        return socialuniImRequest.post<SocialuniImMineUserDetailRO>('socialuni/imUser/openStrangerMsg')
    }
}
